src/providers/details/tabs/Details/components/DetailsSection/onUpdateVddkImageSettings.tsx

```diff
@@ -0,0 +1,64 @@
+import { TRUE_VALUE, YES_VALUE } from 'src/providers/create/utils/constants';
+
+import type { K8sResourceCommon, V1beta1Provider } from '@kubev2v/types';
+import { type K8sModel, k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import { getAnnotations, getSettings } from '@utils/crds/common/selectors';
+import { isEmpty } from '@utils/helpers';
+
+const onUpdateVddkImageSettings = async (
+  model: K8sModel,
+  resource: K8sResourceCommon,
+  {
+    isEmptyImage,
+    isUseVddkAio,
+    newValue,
+  }: { isEmptyImage: boolean; isUseVddkAio: boolean; newValue: string },
+): Promise<K8sResourceCommon> => {
+  const provider = resource as V1beta1Provider;
+  const vddkInitImage: string = newValue;
+
+  // Patch settings
+  const currentSettings = (getSettings(provider) ?? {}) as object;
+  const op1 = Object.keys(currentSettings).length ? 'replace' : 'add';
+  const settings = {
+    ...currentSettings,
+    useVddkAioOptimization: isEmptyImage || !isUseVddkAio ? undefined : TRUE_VALUE,
+    vddkInitImage: isEmptyImage ? undefined : vddkInitImage?.trim() || undefined,
```

Comment:
switch || to ??

Reply:
What about empty string?  the undefined is set for an empty string as well.